packages/core/test/test_util/common_tests.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+'use strict';
+
+const expectExactlyOneMatching = require('./expectExactlyOneMatching');
+const constants = require('../../src/index').tracing.constants;
+
+/**
+ * @typedef {import('../../src/tracing/cls').InstanaBaseSpan} InstanaBaseSpan
+ */
+
+/**
+ * @param {Chai.ExpectStatic} expect
+ * @param {Array.<InstanaBaseSpan>} spans
+ * @param {string} apiPath
+ * @param {string} pid
+ * @param {Array.<(span: InstanaBaseSpan) => void>} [extraTests]
+ * @returns
+ */
+exports.verifyHttpEntry = function verifyHttpEntry(expect, spans, apiPath, pid, extraTests) {
+  /** @type {Array.<(span: InstanaBaseSpan) => void>} */
+  const tests = [
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.p).to.not.exist;
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.k).to.equal(constants.ENTRY);
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.f.e).to.equal(pid);
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.f.h).to.equal('agent-stub-uuid');
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.n).to.equal('node.http.server');
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.data.http.url).to.equal(apiPath);
+    }
+  ].concat(extraTests || []);
+
+  return expectExactlyOneMatching(spans, tests);
+};
+
+/**
+ *
+ * @param {Chai.ExpectStatic} expect
+ * @param {string} spanName
+ * @param {Array.<InstanaBaseSpan>} spans
+ * @param {InstanaBaseSpan} parent
+ * @param {boolean} withError
+ * @param {string} pid
+ * @param {Array.<(span: InstanaBaseSpan) => void>} [extraTests]
+ * @returns
+ */
+exports.verifyExitSpan = function verifyExitSpan(expect, spanName, spans, parent, withError, pid, extraTests) {
+  const tests = [
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.k).to.equal(constants.EXIT);
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.n).to.equal(spanName);
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.t).to.equal(parent.t);
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.p).to.equal(parent.s);
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.f.e).to.equal(pid);
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.f.h).to.equal('agent-stub-uuid');
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.error).to.not.exist;
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.ec).to.equal(withError ? 1 : 0);
+    },
+    (/** @type {InstanaBaseSpan} */ span) => {
+      expect(span.async).to.not.exist;
```

Comment:
This check should be removed completely. I know it was present in all the original verifyXxx methods but it serves no purpose (see my comment in cls.js).